tagstudio/src/qt/ts_qt.py

```diff
@@ -189,9 +186,42 @@ def __init__(self, core, args):
 
         self.SIGTERM.connect(self.handleSIGTERM)
 
-        self.settings = QSettings(
-            QSettings.IniFormat, QSettings.UserScope, "tagstudio", "TagStudio"
-        )
+        if self.args.config_file:
+            path = Path(self.args.config_file)
+            if path.is_dir():
+                path = path / "TagStudio.ini"
+                self.settings = QSettings(str(path), QSettings.IniFormat)
+                logging.info(
+                    f"[QT DRIVER] Directory provided defaulting to TagStudio.ini in directory, using {self.settings.fileName()}"
+                )
+            elif path.is_file():
+                self.settings = QSettings(str(path), QSettings.IniFormat)
+                logging.info(
+                    f"[QT DRIVER] Config File exists, using {self.settings.fileName()}"
+                )
+            else:
+                if path.suffix == ".ini" and path.parent.is_dir():
```

Comment:
I dont understand this condition - `path.parent.is_dir()` will be always true, no?

But I'd say the same as in comment above - just handle the case where a path to existing file is provided, fail loudly otherwise.

Reply:
Not necessarily, 
If config_file is passed a path like `/home/Loran425/projects/TagStudio/TagStudio.ini` the path will always be setup properly from the string but if any of the folders in that chain don't exist then the folder wouldn't exist.

Reply:
I see. In that case `.exists()` would be probably better choice.

Reply:
If we cut out the guess work and the file is expected to already exists yes `.exists()` would 100% be the proper call.

This still cuts out the ability to specify a new config file location from the command line arguments which means a user trying to specify a non-standard location would have to go `touch` the file prior to running TagStudio with a config flag pointing at that location, this doesn't seem ideal.

I think at this point, I'd agree the guess work needs to go, but I'm not sold on the config file existence requirement.